config/env_sample.hjson

```diff
@@ -252,7 +253,34 @@
                 and membership.id is not null 
                 and REGEXP_EXTRACT(`group`.id, r'.*:(.*)') IN UNNEST(@course_ids)
                 '''
-        }
+        },
+        "canvas_pages": {
+            "query":
+                '''
+                SELECT 'canvas_pages' AS resource_type,
+                REPLACE(JSON_EXTRACT_SCALAR(extensions_json, '$[\'com.instructure.canvas\'][request_url]'), 'https://example.instructure.com/courses', '') as resource_id,
+                CAST(REGEXP_EXTRACT(membership.id, r'.*:(.*)') AS INT64) AS user_id,
+                cast(null as string) AS user_login_name,
+                CAST(REGEXP_EXTRACT(`group`.id, r'.*:(.*)') AS INT64) AS course_id,
+                JSON_EXTRACT_SCALAR(object.extensions, '$[\'com.instructure.canvas\'][asset_name]') as name,
+                datetime(EVENT_TIME) as access_time
+                FROM event_store.expanded
+                where
+                JSON_EXTRACT_SCALAR(ed_app.json, '$.id') IN UNNEST(['http://m.example.instructure.com/', 'http://example.instructure.com/'])
+                and type = 'NavigationEvent'
+                and object.type = 'Page'
+                and action = 'NavigatedTo'
+                and membership.id is not null
+                and not CONTAINS_SUBSTR(JSON_EXTRACT_SCALAR(extensions_json, '$[\'com.instructure.canvas\'][request_url]'), '/api/v1/')
+                and REGEXP_EXTRACT(`group`.id, r'.*:(.*)') IN UNNEST(@course_ids)
+                ''',
+            "app_display_name": "Canvas Files",
+            "canvas_course_id_format": "LONG",
+            "urls": {
+                "prefix": "https://example.instructure.com/courses",
```

Comment:
So the url will be something like `https://example.instructure.com/courses/@courseID/pages/@page_id`

Reply:
Right. I think the need to update the prefix url pattern with one more parameter could be problem:

We use the following code inside views.py. to construct resource link for all resource types (file, video, and page). It is hard to add one more param just for the page type.

`    output_df['resource_name'] = output_df.apply(
        lambda row:
            (RESOURCE_ACCESS_CONFIG.get(row.resource_type).get("urls").get("prefix") +
            str(row.r_id) +
            RESOURCE_ACCESS_CONFIG.get(row.resource_type).get("urls").get("postfix") +
            CANVAS_FILE_ID_NAME_SEPARATOR +
            str(row.r_name) + CANVAS_FILE_ID_NAME_SEPARATOR +
            RESOURCE_VALUES.get(RESOURCE_VALUES_MAP.get(row.resource_type)).get('icon')
            ),
        axis=1)`

Reply:
I am ok, if collectively we all agree on this. My Further thoughts on this.

1. So the URL to view page ID https://example.instructure.com/courses/@courseID/pages/@page_id is something Canvas support from Ui if people know it. From the regular Canvas User standpoint, this URL doesn't exist. So there may be a risk that we are constructing this and canvas might discontinue( I don't feel that will happen)
2.  I think having a long URL ( with a descriptive page name) as Id is not ideal. But I really don't know if this is normal elsewhere. **But I want this tested with a very long page name, page name with non-ASCII literals ( like Japanese text) to make sure this approach work**.
3.  I feel in order to support page id as resource id, we might need to re-think about the existing cron design ( with pre-fix and post-fix)

Reply:
Resource id type is `varchar(255)`.

Reply:
Yes, your suggestion is valid. Please check the latest change. 

I have tested the query with courses that have unicode inside page titles. Here is the query that can help locate those courses: 

`
select wpd.title,
    CASE
      WHEN regexp_instr(title, '[^[:print:][:cntrl:]]') > 0 THEN TRUE
      ELSE FALSE END AS has_non_ascii_char,
cd."name",
cd.canvas_id 
from wiki_page_dim wpd, wiki_page_fact wpf, course_dim cd 
where has_non_ascii_char 
and wpf.wiki_page_id =wpd.id
and cd.id = wpf.parent_course_id 
and cd.name like '%SU 2022%'
order by wpf.parent_course_id desc
`

Reply:
Just a quick comment that I'm delighted to see y'all working on this and wish I could have made the contribution myself!

Reply:
@jlongland you are very welcome!